packages/shared/src/porter.ts

```diff
@@ -284,4 +353,89 @@ export class PorterClient {
     > = Object.fromEntries(decryptionResponses);
     return { encryptedResponses, errors };
   }
-}
+
+
+  public async signUserOp(
+    signingRequests: Record<string, string>,
+    threshold: number,
+    options: SigningOptions = {},
+  ): Promise<SignResult> {
+    const data: Record<string, unknown> = {
+      signing_requests: signingRequests,
+      threshold: threshold
+    };
+    
+    const resp: AxiosResponse<SignResponse> = await this.tryAndCall({
+      url: '/sign',
+      method: 'post',
+      data,
+    });
+
+    const { signatures, errors } = resp.data.result.signing_results;
+    const allErrors: Record<string, string> = { ...errors };
+    const { optimistic = false } = options;
+    
+    const signingResults: { [ursulaAddress: string]: [string, string] } = {};
+    let messageHash = '';
+    
+    // For non-optimistic path
```

Comment:
I think there are some tweaks we can make to the logic here, but let's do that in a separate PR.

Thanks for resolving this so quickly!